pkg/skaffold/build/cache/hash_test.go

```diff
@@ -477,3 +477,9 @@ func TestHashBuildArgs(t *testing.T) {
 		})
 	}
 }
+
+type fakeHashOnceImpl struct{}
```

Comment:
Can we add some unit tests for `getHashForArtifact` with dependencies?
e.g. 
1. For  an artifact with same files, same artifact dependencies, in different order produce the same hash
2. For an artifact with same files, different artifact dependencies produce different hash.
3. For  an artifact with same files, same artifact dependency, when an artifact dependency hash changes, the  artifact hash should change

Reply:
Done